### PR TITLE
Transform and load dependencies from setup.cfg

### PR DESCRIPTION
**Pull request by olivia-hong**
Previously, requirements.txt was the only file cartography loaded dependencies from.
This PR adds support for setup.cfg, which is where many Python libraries define dependencies;
specifically in `install_requires`, `setup_requires`, and `extras_require`

Also tested locally by running cartography, ensuring the data was loaded correctly, and checking that I could query dependencies from `setup.cfg`.

Note: the GH dependency graph unfortunately [does not support setup.cfg ingestion](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-the-dependency-graph#supported-package-ecosystems) at the moment. Although setup.cfg support has been implement in [dependabot](https://github.com/dependabot/dependabot-core/pull/3423), the two projects [are completely separate with no timeline for merge](https://github.com/dependabot/dependabot-core/pull/2281#discussion_r516652962) at the moment.

### CODE REVIEW

**Review by crockeo — COMMENTED** [review]
some initial thoughts

File: cartography/intel/github/repos.py
Comment (olivia-hong) [reply cartography/intel/github/repos.py]: The contents are whatever is returned from the graphql API. Right now that's a single key value pair where the value is the entire setup.cfg file as a string, which I don't think is worth defining static typing for.

File: cartography/intel/github/repos.py
Comment (olivia-hong) [reply cartography/intel/github/repos.py]: thanks!

File: cartography/intel/github/repos.py
Comment (olivia-hong) [reply cartography/intel/github/repos.py:277]: optional and changed the function argument, good catch thanks!

Comment (crockeo) [reply cartography/intel/github/repos.py:275]: if you want job security through obfuscation...

`if (text_contents := (setup_cfg_contents or {}).get('text')) is not None:`

jk that's not even valid since we're not guaranteed to be running under python3.8

File: cartography/intel/github/repos.py
Comment (crockeo) [inline cartography/intel/github/repos.py:280]: `import configparser` and `configparser.Error` seems clearer here since `Error` is a fairly generic term

**Review by crockeo — APPROVED** [review]
code changes look good to me. let's also get a review from cartography owners as discussed over slack.

**Review by crockeo — DISMISSED** [review]
re-approving after change and merge

File: tests/data/github/repos.py
Comment (aneeshusa) [inline tests/data/github/repos.py:36]: nit: Some `textwrap.dedent` wouldn't hurt in this file similar to what we do in our internal repos

File: cartography/intel/github/repos.py
Comment (aneeshusa) [reply cartography/intel/github/repos.py:274]: nit: prefer an early return so you can eliminate a level of indentation. i.e.
```
if not setup_cfg_contents or 'text' not in setup_cfg_contents:
    return
text_contents = setup_cfg_contents['text']
...
```

File: cartography/intel/github/repos.py
Comment (aneeshusa) [inline cartography/intel/github/repos.py:130]: What happens if a dep is listed in both places (e.g. no bounds in setup.cfg but pinned in requirements.txt)?
What do we want to have happen (e.g. what would be most useful for our query patterns/the https://github.com/lyft/cartography/blob/master/docs/usage/samplequeries.md sample queries)?
This is likely worth a test case.

File: docs/schema/github.md
Comment (aneeshusa) [inline docs/schema/github.md:178]: Probably worth mentioning that we include everything from all of `install_requires`, `setup_requires`, and `extras_require`, folks might only think about `install_requires` at first.

**Review by aneeshusa — APPROVED** [review]

File: cartography/intel/github/repos.py
Comment (olivia-hong) [reply cartography/intel/github/repos.py:130]: If it's listed in both places and has different specifiers for each usage, cartography will create two separate nodes, which I think makes sense rather than any sort of "merging" logic. This allows users to query what version(s) are being used or perhaps find out that they are specifying a dependency in multiple files when it's not needed.

Comment (aneeshusa) [reply cartography/intel/github/repos.py:130]: That SGTM, thanks! Just wanted to check that we didn't have one "overwrite" the other

File: cartography/intel/github/repos.py
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [inline cartography/intel/github/repos.py:258]: Would prefer this named the same as the param in the function you  call below

**Comment by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg** [comment]
Re testing, have you run this on a few setup.cfgs from some of  our bigger projects?

**Review by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg — DISMISSED** [review]
Looks good; a few minor questions but good to go.

**Comment by olivia-hong** [comment]
> Re testing, have you run this on a few setup.cfgs from some of our bigger projects?

I tested on repos with setup.cfg's that had requirements across install_requires, setup_requires, and extras_require.

**Review by aneeshusa — APPROVED** [review]

**Comment by olivia-hong** [comment]
🚀

**Pull request merged**
